sdk/tests/deqp/framework/common/tcuTestCase.js

```diff
@@ -161,9 +161,50 @@ goog.scope(function() {
     tcuTestCase.DeqpTest.prototype.init = function() {};
 
     /**
-     * Abstract deinit function(each particular test will implement it, or not)
+     * Deinit
      */
-    tcuTestCase.DeqpTest.prototype.deinit = function() {};
+    tcuTestCase.DeqpTest.prototype.deinit = function() {
+        // Multisampling state
+        gl.disable(gl.SAMPLE_COVERAGE);
+        gl.disable(gl.SAMPLE_ALPHA_TO_COVERAGE);
+
+        // Texture state
+        gl.bindTexture(gl.TEXTURE_2D, null);
```

Comment:
This is unlikely to be sufficient, given that at least some of the tests call activeTexture(). I'm not sure whether the best thing to do would be to iterate through all of the texture units, or remove this from here and do it in the deinit() methods of test subclasses that need it.